Services/User/classes/class.ilUserAppEventListener.php

```diff
@@ -0,0 +1,20 @@
+<?php declare(strict_types=1);
+
+/**
+ * Class ilUserAppEventListener
+ * @author Michael Jansen <mjansen@databay.de>
+ */
+class ilUserAppEventListener implements ilAppEventListener
+{
+    /**
+     * @inheritDoc
+     */
+    public static function handleEvent($a_component, $a_event, $a_parameter)
+    {
+        if ('Services/Object' === $a_component && 'beforeDeletion' === $a_event) {
+            if (isset($a_parameter['object']) && $a_parameter['object'] instanceof ilObjRole) {
+                \ilStartingPoint::onRoleDeleted($a_parameter['object']);
```

Comment:
Why is this listener located in Services/User instead of Services/AccessControl where ilStartingPoint is located?

Reply:
Because `ilUserStartingPointGUI` and `ilUserRoleStartingPointTableGUI` are located in `./Services/User`.

But of course I could change the location for the listener class according to `ilStartingPoint`. Please advise.

Reply:
:-) Ok, keep it there.